File: 01.Find lower bound.cpp
int lowerBound(vector<int> arr, int n, int x) {
	// Write your code here
    // lower bound is the smallest index i such that arr[i] >= num
    int lower_bound = n ;
    int l = 0,r=n-1;
    while(l<=r){
        int mid = (l+ (r-l)/2);
        // probable answer go left 
        if(arr[mid]>=x){
            lower_bound = mid;
            r = mid - 1;
        }
        else {
            l = mid+1;
        }
    }
    return lower_bound;
}

File: Binary Search/01.Find_Lower_Bound.js
class Solution {
    lowerBound(arr, target) {
        // code here
        // lower bound is the smallest i such that arr[i]>=target
        const n = arr.length;
        let [l,r] = [0,n-1];
        let res = n;
        while(l<=r){
            const mid = Math.floor(l+(r-l)/2);
            if(arr[mid]< target) l = mid+1;
            else {
                res = mid;
                r = mid-1;
            }
        }
        return res;
    }
}
